includes/Modules/Search_Console/Email_Reporting/Report_Data_Processor.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Search_Console\Email_Reporting\Report_Data_Processor
+ *
+ * @package   Google\Site_Kit\Modules\Search_Console\Email_Reporting
+ * @copyright 2025 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Search_Console\Email_Reporting;
+
+/**
+ * Processes Search Console data for email reporting (sorting, partitioning, summarizing).
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Report_Data_Processor {
+
+	/**
+	 * Sorts Search Console rows by a given field.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array  $rows  Search Console rows (arrays or objects).
+	 * @param string $field Field name such as 'ctr', 'clicks', etc.
+	 * @param string $order Optional. 'asc' or 'desc'. Default 'desc'.
+	 * @return array Sorted rows.
+	 */
+	public function sort_rows_by_field( array $rows, $field, $order = 'desc' ) {
+		$direction = ( 'asc' === strtolower( $order ) ) ? 'asc' : 'desc';
+
+		usort(
+			$rows,
+			function ( $a, $b ) use ( $field, $direction ) {
+				$value_a = $this->extract_row_value( $a, $field );
+				$value_b = $this->extract_row_value( $b, $field );
+
+				if ( $value_a === $value_b ) {
+					return 0;
+				}
+
+				if ( 'asc' === $direction ) {
+					return ( $value_a < $value_b ) ? -1 : 1;
+				}
+
+				return ( $value_a > $value_b ) ? -1 : 1;
+			}
+		);
+
+		return $rows;
+	}
+
+	/**
+	 * Partitions rows into compare and current periods.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $rows          Combined-period rows returned from the API.
+	 * @param int   $period_length Number of days within a period.
+	 * @return array {
+	 *     @type array $compare Rows for the earlier period.
+	 *     @type array $current Rows for the current period.
+	 * }
```

Comment:
Same here. Let's use something simpler, like this:
```suggestion
	 * @return array An array with two sets of rows: one for the current range and one for the compare date range.
```